catchup/service.go

```diff
@@ -557,20 +554,16 @@ func (s *Service) syncCert(cert *PendingUnmatchedCertificate) {
 // TODO this doesn't actually use the digest from cert!
 func (s *Service) fetchRound(cert agreement.Certificate, verifier *agreement.AsyncVoteVerifier) {
 	blockHash := bookkeeping.BlockHash(cert.Proposal.BlockDigest) // semantic digest (i.e., hash of the block header), not byte-for-byte digest
-	fetcher := s.latestRoundFetcherFactory.NewOverGossip()
-	defer func() {
-		fetcher.Close()
-	}()
+	peerSelector := s.createPeerSelector(false)
 	for s.ledger.LastRound() < cert.Round {
-		if fetcher.OutOfPeers(cert.Round) {
-			fetcher.Close()
-			// refresh peers and try again
-			logging.Base().Warn("fetchRound found no outgoing peers")
-			s.net.RequestConnectOutgoing(true, s.ctx.Done())
-			fetcher = s.latestRoundFetcherFactory.NewOverGossip()
+		peer, getPeerErr := peerSelector.GetNextPeer()
+		if getPeerErr != nil {
+			s.log.Debugf("fetchRound: was unable to obtain a peer to retrieve the block from")
+			break
```

Comment:
the `fetchRound` method need to ensure that it doesn't exit before the round is fetched. Therefore, in this case, you want to call `s.net.RequestConnectOutgoing(true, s.ctx.Done())` and `continue`. The `peerSelector.GetNextPeer` would internally try to pick new relays.